### PR TITLE
first steps

### PR DESCRIPTION
**Pull request by KhafraDev**
Adds basic parsing for mime types' type (ie. the type of `text/plain` is `text`). Just making sure everything is fine, which it probably isn't, before continuing.

### CODE REVIEW

File: src/parser.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include "util.cpp"
+
+#include <string>
+
+namespace ada::mimesniff {
+void parse_mime_type (std::string_view input) {
+  // step 1
+  ada::mimesniff::trim_http_whitespace(input);
```
Comment (anonrig) [inline src/parser.cpp]:
```suggestion
  trim_http_whitespace(input);
```

File: src/parser.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  // step 5
+  if (position == std::string_view::npos) {
+    return;
+  }
+
+  // step 3
+  std::string_view type = input.substr(0, position);
+
+  // step 4
+  if (type.empty() ||
```
Comment (anonrig) [inline src/parser.cpp]: you can move this type.empty check to previous if statement.

File: src/parser.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      !ada::mimesniff::contains_only_http_tokens(type)) {
+    return;
+  }
+
+  // step 6
+  position++;
```
Comment (anonrig) [inline src/parser.cpp:27]: why do you increment this?

Comment (KhafraDev) [reply src/parser.cpp:27]: `Advance position by 1. (This skips past U+002F (/).) `, https://mimesniff.spec.whatwg.org/#parsing-a-mime-type step 6

Comment (KhafraDev) [reply src/parser.cpp:27]: eventually we'll need position for the start of the subtype

**Pull request merged**